sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -534,17 +579,11 @@ class SessionCatalog(
     val dbFunctions =
       externalCatalog.listFunctions(db, pattern).map { f => FunctionIdentifier(f, Some(db)) }
     val regex = pattern.replaceAll("\\*", ".*").r
```

Comment:
I may create another PR for put all the pattern stuff in one place. Right now, it was scattered all over with replaceAll(). Please keep your code as it is for now, I will do it later. Thanks.